File: practica3_820574_839304/c++/inventario.cc
/*
* Fichero: inventario.cc
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 09/03/2024
*/

#include "inventario.h"

Inventario::Inventario(double _volumen)
                :volumen(_volumen)
{
    nivel = 0;
}

string Inventario::nombre() const
{
    return name;
}

string Inventario::get_descripcion() const
{
    return nombre() + " [" + to_string(get_volumen()) + " m3] [" + to_string(get_peso()) + " kg]";
}

int Inventario::get_nivel() const
{
    return nivel;
}

double Inventario::get_volumen() const
{
    return volumen;
}

File: practica3_820574_839304/java/Inventario.java
/*
* Fichero: Inventario.java
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 29/02/2024
*/

abstract class Inventario
{
    protected String nombre;
    protected double volumen;

    Inventario(double _volumen)
    {
        volumen = _volumen;
    }

    public String nombre()
    {
        return nombre;
    }

    public double volumen()
    {
        return volumen;
    }

    abstract public double peso();

    abstract public String toString();
}